### PR TITLE
new crustal deformation cookbook

### PR DESCRIPTION
**Pull request by cedrict**

### CODE REVIEW

File: cookbooks/crustal_model_2D.prm [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# this is a version of the stokes' law input file
+# but with less mesh refinement
```
Comment (bangerth) [inline cookbooks/crustal_model_2D.prm]: Comment is not up to date.

**Comment by QwZhang** [comment]
Thank you for your new Drucker-Prager model, which enable ASPECT to deal with materials with plasticity, which is common for rocks in the lithosphere. Could it be possible to add a strain-weakening/softening capability to this model, in which one can extend your Drucker-Prager model by configuring parameter like set enable strain weaking = true/false etc.? In a previous example, this was achieved by linearly decreasing angle of internal friction ϕ from 15° to 4°over the range of strain, where 15° and 4° represents a none and fully strain-softened material, respectively (C. Beaumont, 2004). Thank you!

**Comment by bangerth** [comment]
@QwZhang , as mentioned on the mailing list before, open source projects rely on community participation. We would definitely appreciate if you would implement the suggestions you mention in your comment and contributed them to the project!

File: doc/manual/manual.tex [hunk 2]
```diff
@@ -5558,6 +5558,128 @@ \subsubsection{Reproducing Rheology of Morency and Doin, 2004}
...
+
+
+
+
+
+\subsubsection{Crustal deformation}
+
+
+\textit{This section was contributed by Cedric Thieulot, and makes use of the Drucker-Prager 
+material model written by Anne Glerum and the free surface plugin by Ian Rose.}
```
Comment (bangerth) [inline doc/manual/manual.tex]: nice way to give credit!

File: doc/manual/manual.tex [hunk 3]
```diff
@@ -5558,6 +5558,128 @@ \subsubsection{Reproducing Rheology of Morency and Doin, 2004}
...
+
+\begin{figure}
+  \centering
+  \includegraphics[width=\textwidth]{cookbooks/crustal_deformation/bottombc2}
+  \caption{Basal velocity boundary conditions and corresponding 
+  strain rate field for the 3D model.} 
+  \label{bottombc}
+\end{figure}
+
+The Finite Element mesh, the velocity, viscosity and strain rate fields are shown
```
Comment (bangerth) [inline doc/manual/manual.tex]: use lower-case letters for "finite element".

**Comment by cedrict** [comment]
Hi,
thank you for your message. I am afraid that strain softening/weakening will not be implemented any time soon.
To this purpose one could use an additional composition, or the soon to be committed tracer library.
I agree with you that this would be a nice feature and encourage you to help us out implementing it.

Best,

Cedric.

**Comment by QwZhang** [comment]
@bangerth you are right and it has been on my to-do list. I used other codes like citcoms and underworld/Gale and know what they can do and what cannot. Problem is that most of them are poorly documented to get involved in and thus is actually not very easy to use. So I switch to Aspect because I like the way Aspect is develped and happy to see that it has an very active community and straightforward interfaces are provided to enable easy expansion of its capabilities. Unfortunately I am so far not very familiar with the whole workflow of How to get involved in (e.g., testing). So perhaps experences and time are what matters. Thank you for what you have done.
Best,
Qingwen

**Comment by jdannberg** [comment]
Very cool, thanks!

**Pull request merged**